test/playwright/utils/navigation.ts

```diff
@@ -11,7 +11,57 @@ import {
 } from '~/constants/media'
 import { SCREEN_SIZES } from '~/constants/screens'
 
-import messages from '~/locales/en.json'
+import enMessages from '~/locales/en.json'
+import rtlMessages from '~/locales/ar.json'
+
+const messages: Record<string, Record<string, unknown>> = {
+  ltr: enMessages,
+  rtl: rtlMessages,
+}
+
+const getNestedProperty = (
+  obj: Record<string, unknown>,
+  path: string
+): string => {
+  const value = path
+    .split('.')
+    .reduce((acc: string | Record<string, unknown>, part) => {
+      if (typeof acc === 'string') {
+        return acc
+      }
+      if (Object.keys(acc as Record<string, unknown>).includes(part)) {
+        return (acc as Record<string, string | Record<string, unknown>>)[part]
+      }
+      return ''
+    }, obj)
+  return typeof value === 'string' ? value : JSON.stringify(value)
+}
+
+/**
+ * Simplified i18n t function that returns English messages for `ltr` and Arabic for `rtl`.
+ * It can also handle nested labels ('header.title').
+ * @param path - The label to translate.
+ * @param dir - The language direction.
+ */
+export const t = (path: string, dir: LanguageDirection = 'ltr'): string => {
+  let value = ''
+  if (dir === 'rtl') {
+    value = getNestedProperty(messages.rtl, path)
+  }
+  return value === '' ? getNestedProperty(messages.ltr, path) : value
+}
```

Comment:
I love this!!!

I actually started to write something like this when I was working on the flaky audio test fix but realized I didn't need it then, I'm so glad we're getting it for real!

Question: should the case where the string doesn't match raise an error, by any chance? :thinking: I guess it could interrupt the `ar` one if we're not consistently updating it with the new strings?

Reply:
> Question: should the case where the string doesn't match raise an error, by any chance?

Sorry, I don't fully understand this question. What should the string match? I don't think we can compare to previous versions here, because we don't have the previous versions of the locales...
I mimicked the behavior of Vue-i18n here so that if the locale doesn't have the key, it falls back to `en.json` value.

Reply:
Oh I see! I misread the code. Looks great then, never mind :slightly_smiling_face: